doc_files/service_tutorial.md

```diff
@@ -0,0 +1,119 @@
+
+## Step 1: Deploy the bookinfo Application
+
+Create a new namespace: $NAMESPACE. We use the name `bookinfo-serivce`.
+
+```bash
+export NAMESPACE=bookinfo-service
+kubectl create ns $NAMESPACE
+kubectl label ns $NAMESPACE istio-injection=enabled
+```
+
+Deploy the bookinfo application to a new namespace. In particular, we create the service _productpage-v1_ to access the _productpage_ application.
+
+```bash
+kubectl -n $NAMESPACE apply -f https://raw.githubusercontent.com/iter8-tools/iter8-controller/v0.2/doc/tutorials/istio/bookinfo/bookinfo-tutorial.yaml -f https://raw.githubusercontent.com/iter8-tools/iter8-controller/v0.2/doc/tutorials/istio/bookinfo/service/productpage-v1.yaml
+```
+
+Create an Istio gateway for the external host `productpage.example.com`:
+
+```bash
+kubectl -n $NAMESPACE apply -f https://raw.githubusercontent.com/iter8-tools/iter8-controller/v0.2/doc/tutorials/istio/bookinfo/service/bookinfo-gateway.yaml
+```
+
+******
+Note that at this point, the application is not actually accessible because no Istio `VirtualService` is defined. This tutorial shows how iter8 will create the `VirtualService` for us. This is a use case that might apply, for example, when we do not specify all the Istio components in an application such as when using `helm`.
+
+## Step 2: Create an iter8 Canary Experiment
+
+We can now create an iter8 `Experiment` specifying using the Kubernetes services. The `targetService` portion of the `Experiment` look like this:
+
+```yaml
+targetService:
+    kind: Service
+    baseline: productpage-v1
+    candidate: productpage-v2
+    hosts:
+      - name: productpage.example.com
+        gateway: productpage-service
+```
+
+It identifies the type of the baseline and candidate as services using `kind: Service`. Finally, it identifies the external host name and the Istio `Gateway` already configured to route traffic.
+
+You can create the `Experiment` using:
+
+```bash
+kubectl -n $NAMESPACE apply -f https://raw.githubusercontent.com/iter8-tools/iter8-controller/v0.2/doc/tutorials/istio/bookinfo/service/canary_productpage-v1_to_productpage-v2.yaml
+```
+
+You can verify that the `Experiment` has been created:
+
+```bash
+kubectl -n $NAMESPACE get experiment productpage-v2-rollout
+NAME                     PHASE   STATUS                               BASELINE         PERCENTAGE   CANDIDATE        PERCENTAGE
+productpage-v2-rollout   Pause   TargetsNotFound: Missing Candidate   productpage-v1   100          productpage-v2   0
+```
+
+The experiment is paused since only the baseline version can be identified. When the candidate version is detected, the experiment will automatically begin execution.
+
+## Step 3: Generate load
+
+As in earlier tutorials, emulate requests coming from users using `curl`:
+
+```bash
+watch -x -n 0.1 curl -Is -H 'Host: productpage.example.com' "http://${GATEWAY_URL}/productpage"
```

Comment:
Maybe the `/productpage` uri can be removed, as no uri is specified. Although this curl works, it looks strange that a new path is added.

Reply:
@kimikowang would this require a change to the gateway? Currently, if I remove the path, I do not get the product page application. I get a page that says "Hello! This is a simple bookstore application consisting of three services as shown below"

Reply:
I see. I forgot about the nested path in the application. Nevermind.